assembly/src/errors.rs

```diff
@@ -438,6 +465,53 @@ impl fmt::Display for ParsingError {
 #[cfg(feature = "std")]
 impl std::error::Error for ParsingError {}
 
+// PROCEDURE NAME ERROR
+// ================================================================================================
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub enum ProcedureNameError {
```

Comment:
I'm hoping that we can keep the number of such variants low. Here, since a procedure name error could happen in many places, I didn't want to make it a part of other errors. Same could apply maybe to a couple more instances.